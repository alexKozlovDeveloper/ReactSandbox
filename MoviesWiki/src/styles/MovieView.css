.MovieView{
    
}

.body {
    margin: 0 auto;
    background-image: url(http://hd-wallpapers.co/wp-content/uploads/walls/thumbs/Dark-Blue-1024x640.jpg);
    background-size: contain;
    min-width: 500px;
    min-height: calc(100vh - 163px);
    padding-left: 30px;
    padding-right: 30px;
    padding-top: 10px;
    padding-bottom: 10px;
}

.infocontainer {

}

.genretitle {
    float: left;
    color: white;
    margin-left: 15px;
    margin-bottom: 5px;
    margin-top: 14px;
}
