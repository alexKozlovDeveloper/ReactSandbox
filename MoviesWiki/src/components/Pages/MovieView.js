import React, { Component } from "react";
import { connect } from 'react-redux'

import DetailsView from "../DetailsView";
import ItemsContainer from "../Container/ItemsContainer";

import Footer from "../Footer";
import Spinner from "../Common/Spinner";

import styles from "../../styles/MovieView.css"

import { updateMovies, loading } from "../../actions/index";

class MovieView extends Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        //debugger;
        let genres = this.props.selectedItem.genres;

        if (genres.lenght < 1) {
            return;
        }

        let genre = genres[0]

        let url = "https://reactjs-cdp.herokuapp.com/movies?search=" + genre + "&searchBy=genres&sortBy=vote_average&sortOrder=asc"

        this.props.loadingFunc()
 
        fetch(url)
            .then(res => res.json())
            .then(
                (result) => {
                    //debugger;
                    this.props.updateMoviesFunc(result);
                },
                (error) => {
                    //debugger;
                    // TODO: Process error
                }
            )

    }

    render() {

        if (this.props.isLoaded === false || this.props.movies == null) {
            return (
                <div className={styles.body}>
                    <Spinner></Spinner>
                </div>)
        }

        let genres = this.props.selectedItem.genres;

        if (genres.lenght < 1) {
            return;
        }

        let genre = genres[0];

        return (
            <div className={styles.MovieView}>
                <DetailsView />
                <div className={styles.body}>
                    <div className={styles.infocontainer}>
                        <div className={styles.genretitle}>
                            Films by {genre} genre
                    </div>
                    </div>
                    <div>
                        <ItemsContainer items={this.props.movies} />
                    </div>
                </div>
                <Footer config={this.props.footerConfig} />
            </div>)
    }
}


function mapDispatchToProps(dispatch) {
    return {
        updateMoviesFunc: (movies) => {
            dispatch(updateMovies(movies.data))
        },
        loadingFunc: () => {
            dispatch(loading())
        }
    }
}

function mapStateToProps(state) {
    const { movies, isLoaded, sortBy, selectedItem } = state.moviesReducer;

    return {
        selectedItem: selectedItem,
        movies: movies,
        isLoaded: isLoaded,
        sortBy: sortBy
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(MovieView);
