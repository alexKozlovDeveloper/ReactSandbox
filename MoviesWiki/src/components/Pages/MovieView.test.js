import React from 'react';
import { mount, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {
  Response, Request, Headers, fetch,
} from 'whatwg-fetch';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import { BrowserRouter as Router } from 'react-router-dom';
import rootReducer from '../../reducers';

import MovieView from './MovieView';

configure({ adapter: new Adapter() });

const moviesList = [{
  title: 'Interstellar',
  releaseDate: '2014',
  image: 'https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg',
  genre: 'action',
  rating: 7.2,
  duration: 145,
  description: 'is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.',
}, {
  title: 'Interstellar 2',
  releaseDate: '2015',
  image: 'https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg',
  genre: 'action2',
  rating: 7.2,
  duration: 145,
  description: 'is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.',
}, {
  title: 'Interstellar 3',
  releaseDate: '2016',
  image: 'https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg',
  genre: 'action 3',
  rating: 7.2,
  duration: 145,
  description: 'is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.',
}];

const selectedItem = {
  title: 'Interstellar',
  releaseDate: '2014',
  image: 'https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg',
  genres: ['action'],
  rating: 7.2,
  duration: 145,
};

test('MovieView rendering test', () => {
  const defaultStore = {
    moviesReducer: {
      movies: moviesList,
      isLoaded: true,
      selectedItem,
    },
  };

  global.Response = Response;
  global.Request = Request;
  global.Headers = Headers;
  global.fetch = fetch;

  const match = {
    params: {
      id: '99',
    },
  };

  const store = createStore(rootReducer, defaultStore);

  const component = mount(<Router><Provider store={store}><MovieView match={match}/></Provider></Router>);
  expect(shallowToJson(component)).toMatchSnapshot();
});

test('MovieView spinner rendering test', () => {
  const defaultStore = {
    moviesReducer: {
      movies: null,
      isLoaded: false,
      selectedItem,
    },
  };

  global.Response = Response;
  global.Request = Request;
  global.Headers = Headers;
  global.fetch = fetch;

  const store = createStore(rootReducer, defaultStore);

  const match = {
    params: {
      id: '99',
    },
  };

  const component = mount(<Router><Provider store={store}><MovieView match={match}/></Provider></Router>);
  expect(shallowToJson(component)).toMatchSnapshot();
});
