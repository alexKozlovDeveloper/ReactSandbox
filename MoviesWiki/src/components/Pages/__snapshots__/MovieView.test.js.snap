// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieView rendering test 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Connect(MovieView)
        match={
          Object {
            "params": Object {
              "id": "99",
            },
          }
        }
      >
        <MovieView
          isLoaded={false}
          loadingFunc={[Function]}
          match={
            Object {
              "params": Object {
                "id": "99",
              },
            }
          }
          movies={Array []}
          selectedItem={null}
          sortBy="vote_average"
          updateMoviesFunc={[Function]}
          updateSelectedItemFunc={[Function]}
        >
          <div
            className="body"
          >
            <Spinner>
              <div
                className="spinnercontainer"
              >
                <div
                  className="spinner"
                >
                  <div />
                  <div />
                  <div />
                </div>
              </div>
            </Spinner>
          </div>
        </MovieView>
      </Connect(MovieView)>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`MovieView spinner rendering test 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Connect(MovieView)
        match={
          Object {
            "params": Object {
              "id": "99",
            },
          }
        }
      >
        <MovieView
          isLoaded={false}
          loadingFunc={[Function]}
          match={
            Object {
              "params": Object {
                "id": "99",
              },
            }
          }
          movies={Array []}
          selectedItem={null}
          sortBy="vote_average"
          updateMoviesFunc={[Function]}
          updateSelectedItemFunc={[Function]}
        >
          <div
            className="body"
          >
            <Spinner>
              <div
                className="spinnercontainer"
              >
                <div
                  className="spinner"
                >
                  <div />
                  <div />
                  <div />
                </div>
              </div>
            </Spinner>
          </div>
        </MovieView>
      </Connect(MovieView)>
    </Provider>
  </Router>
</BrowserRouter>
`;
